project/src/store/reducer.ts

```diff
@@ -0,0 +1,29 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { Film, FilmGenre } from '../types/films';
+import { setFilmsByGenre, setGenre } from './action';
+import { films } from '../mocks/films';
+import { ALL_GENRES } from '../const';
+
+type initialStateType = {
+  currentGenre: FilmGenre;
+  filmsByGenre: Film[] | []
```

Comment:
```suggestion
  filmsByGenre: Film[];
```

так нужно. А то ты говоришь, что или массив с любыми данными